libs/blocks/faas/utils.js

```diff
@@ -0,0 +1,302 @@
+            'Campos obligatorios',
+            'Obligatoriska fält',
+            'Pola wymagane',
+            'Обязательные поля',
+            'Os campos obrigatórios',
+            '필수 입력란',
+            '为必填项',
+            '為必填項',
+            'Zorunlu alanlar',
+          ];
+          const validnote = $.each(notearr, (i, note) => {
+            const text = note.textContent;
+            const notematch = RMTEXT.indexOf(text);
+            if (notematch >= 0) {
+              return note;
+            }
+            return undefined;
+          });
+
+          return validnote;
+        }
+
+        function childListMutation(cb) {
+          return new MutationObserver((mutations) => {
+            cb(mutations, this);
+          });
+        }
+
+        function setMutationObserver(elms = [], mutation) {
+          $.each(elms, (i, elm) => {
+            mutation.observe(elm, {
+              childList: true
+            });
+          });
+        }
+
+        changeSelectionElement();
+        removeRequired();
+        placeHolders();
+
+        const errorMessages = $('.hash-window .checkbox .errorMessage');
+        const faasform = $('.faas-form');
+        const note = requireNote();
+
+        if (note) {
+          note.remove();
+        }
+
+        if (state['hidePrepopulated']) {
+          const prepop = $('.prepopulated', wr);
+          hideDisplay(prepop);
+        }
+
+        $.each(faasform, (i, form) => {
+          $(form).on('change', () => {
+            const elements = $('span.required', form);
+            hideDisplay(elements);
+          });
+        });
+
+        $('.faas-form').on('submit', (event) => {
+          const firstError = event.target.querySelector('.error [name]');
+          const subNavEl = $('.Subnav-wrapper');
+          const elHeight = subNavEl && subNavEl[0] ? subNavEl[0].offsetHeight : 0;
+
+          if (firstError) {
+            const pageTop = document.body.getClientRects()[0].top;
+            const firstErrorTop = firstError.getClientRects()[0].top;
+            const relPostion = pageTop - firstErrorTop;
+
+            $(firstError).on('focus', (ev) => {
+              ev.preventDefault();
+
+              window.scrollTo(0, (Math.abs(relPostion) - (elHeight * 2.5)));
+            });
+
+            firstError.focus();
+          }
+        });
+
+        setMutationObserver(errorMessages, childListMutation(editMessages));
+      },
+      afterSubmitCallback: (formData) => {
+        // Gate Stuff
+        const disableAutoResponseChk = typeof disableAutoResponse === 'string';
+        const closeModalElement = modal ? modal.querySelector('.dexter-CloseButton') : undefined;
+        const isSameDestination = typeof sameDestination === 'string';
+
+        if (disableAutoResponseChk && state['isGate']) {
+          blocker.parentElement.removeChild(blocker);
+        }
+
+        if (disableAutoResponseChk && isSameDestination && isinmodal && closeModalElement) {
```

Comment:
The code for `afterYiiLoadedCallback` and `afterSubmitCallback`  functions is copied from Dexter. I did that because I don't know what they are exactly but won't hurt to have it. Definitely, we want to revisit them before we ship this,
but I want to leave it there for now.